lib/spack/spack/compilers/clang.py

```diff
@@ -57,6 +57,10 @@ class Clang(Compiler):
         # compilers.yaml
         link_paths['f77'] = 'clang/gfortran'
         link_paths['fc'] = 'clang/gfortran'
+    elif spack.architecture.sys_type() == 'linux-rhel7-ppc64le':
+        # This platform uses clang with IBM XL Fortran compiler
```

Comment:
Are other combinations possible there? Maybe someone installs flang?

Put it at the end of if-else if?

Reply:
@tgamblin I guess there is no way currently to have compiler wrappers pretend to be `xl` for this `clang+xl` combo while still allowing to use `clang+flang` on `linux-rhel7-ppc64le`. 

I have no idea about usage cases, but if it's the common usage (like `clang`+`gfortran` on macOS), i think it's ok.